drivers/gpio/gpio_sedi.c

```diff
@@ -0,0 +1,329 @@
+/* Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_sedi_gpio
+
+#include "sedi_driver_gpio.h"
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/pm/device.h>
+
+struct gpio_sedi_config {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_config common;
+	sedi_gpio_t device;
+	uint32_t pin_nums;
+	void (*irq_config)(void);
+};
+
+struct gpio_sedi_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_config common;
+	sys_slist_t callbacks;
+};
+
+static int gpio_sedi_init(const struct device *dev);
+
+#ifdef CONFIG_PM_DEVICE
+static int gpio_sedi_suspend_device(const struct device *dev)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+	int ret;
+
+	if (pm_device_is_busy(dev)) {
+		return -EBUSY;
+	}
+
+	ret = sedi_gpio_set_power(gpio_dev, SEDI_POWER_SUSPEND);
+
+	if (ret != SEDI_DRIVER_OK) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int gpio_sedi_resume_device_from_suspend(const struct device *dev)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+	int ret;
+
+	ret = sedi_gpio_set_power(gpio_dev, SEDI_POWER_FULL);
+	if (ret != SEDI_DRIVER_OK) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int gpio_sedi_pm_action(const struct device *dev,
+			      enum pm_device_action action)
+{
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		ret = gpio_sedi_suspend_device(dev);
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		ret = gpio_sedi_resume_device_from_suspend(dev);
+		break;
+
+	default:
+		ret = -ENOTSUP;
+	}
+
+	return ret;
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static void gpio_sedi_callback(const uint32_t pin_mask,
+			       const sedi_gpio_port_t port,
+			       void *param)
+{
+	ARG_UNUSED(port);
+	struct device *dev = (struct device *)param;
+	struct gpio_sedi_data *data =
+		(struct gpio_sedi_data *)(dev->data);
+
+	/* call the callbacks */
+	gpio_fire_callbacks(&data->callbacks, dev, pin_mask);
+}
+
+static void gpio_sedi_write_raw(const struct device *dev,
+				uint32_t pins,
+				bool is_clear)
+{
+	uint8_t i;
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+	sedi_gpio_pin_state_t val;
+
+	if (is_clear) {
+		val = SEDI_GPIO_STATE_LOW;
+	} else {
+		val = SEDI_GPIO_STATE_HIGH;
+	}
+
+	for (i = 0; i < config->pin_nums; i++) {
+		if (pins & 0x1) {
+			sedi_gpio_write_pin(gpio_dev, i, val);
+		}
+		pins >>= 1;
+		if (pins == 0) {
+			break;
+		}
+	}
+}
+
+static int gpio_sedi_configure(const struct device *dev, gpio_pin_t pin,
+			       gpio_flags_t flags)
+{
```

Comment:
it seems there're more other gpio config options, return error for them if not support.

Reply:
The only must config is direction, others can be optional, no need to check

Reply:
from gpio.h, the flags only have these settings